numpy/lib/arraysetops.py

```diff
@@ -261,6 +261,36 @@ def unique(ar, return_index=False, return_inverse=False,
         ret = _unique1d(ar, return_index, return_inverse, return_counts)
         return _unpack_tuple(ret)
 
+    if ar.size == 0:
+        # if there's no elements (at least one axis is 0), the reshaping and
+        # viewing doesn't work, but there's also no data, so those
+        # manipulations aren't needed to get right answer.
+
+        axis_len = ar.shape[axis]
+        if axis_len == 0:
+            # there's no elements along this axis, so there's no unique
+            # elements:
+            num_empties = return_index + return_inverse + return_counts
+            output = (ar,) + (np.array([], dtype=np.intp),) * num_empties
```

Comment:
I guess returning the same array should be OK for an empty one...

Reply:
I think the dtype and shape may be different between these values and `ar`.